homeassistant/components/sensor/dht.py

```diff
@@ -112,18 +123,20 @@ def unit_of_measurement(self):
     def update(self):
         """Get the latest data from the DHT and updates the states."""
         self.dht_client.update()
+        temperature_offset = self.temperature_offset
+        humidity_offset = self.humidity_offset
         data = self.dht_client.data
 
         if self.type == SENSOR_TEMPERATURE:
             temperature = round(data[SENSOR_TEMPERATURE], 1)
             if (temperature >= -20) and (temperature < 80):
-                self._state = temperature
+                self._state = round(temperature + temperature_offset, 1)
```

Comment:
This time I got your point. I removed the first round for temperature and humidity and added a _LOGGER.debug it might help in case of problems.